src/Estimators/src/SubModelKinDynWrapper.cpp

```diff
@@ -0,0 +1,328 @@
+    }
+
+    for (int idx = 0; idx < m_subModel.getGyroscopeList().size(); idx++)
+    {
+        Eigen::MatrixXd tempJacobian(6, 6 + m_numOfJoints);
+        tempJacobian.setZero();
+        m_jacGyroList[m_subModel.getGyroscopeList()[idx].name] = std::move(tempJacobian);
+    }
+
+    for (int idx = 0; idx < m_subModel.getExternalContactList().size(); idx++)
+    {
+        Eigen::MatrixXd tempJacobian(6, 6 + m_numOfJoints);
+        tempJacobian.setZero();
+        m_jacContactList[m_subModel.getExternalContactList()[idx]] = std::move(tempJacobian);
+    }
+
+    return true;
+}
+
+bool RDE::SubModelKinDynWrapper::updateKinDynState(
+    std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel)
+{
+    constexpr auto logPrefix = "[BipedalLocomotion::Estimators::SubModelKinDynWrapper::"
+                               "updateKinDynState]";
+
+    // Get model base position and velocity
+    const std::string baseFrameName = m_kinDyn.getFloatingBase();
+
+    // Get world transform for the base frame
+    iDynTree::Transform worldTBase = kinDynFullModel->getWorldTransform(baseFrameName);
+
+    // Get base velocity
+    iDynTree::Twist baseVelocity = kinDynFullModel->getFrameVel(baseFrameName);
+
+    // Get position and velocity of the joints in the full model
+    iDynTree::VectorDynSize jointPositionModel(kinDynFullModel->getNrOfDegreesOfFreedom());
+    iDynTree::VectorDynSize jointVelocityModel(kinDynFullModel->getNrOfDegreesOfFreedom());
+    iDynTree::Vector3 worldGravity;
+    kinDynFullModel->getRobotState(jointPositionModel, jointVelocityModel, worldGravity);
+
+    // Get positions and velocities of joints of the sub model
+    iDynTree::VectorDynSize qj(m_subModel.getModel().getNrOfDOFs()),
+        dqj(m_subModel.getModel().getNrOfDOFs());
+
+    for (int idx = 0; idx < m_subModel.getJointMapping().size(); idx++)
+    {
+        qj.setVal(idx, jointPositionModel[m_subModel.getJointMapping()[idx]]);
+        dqj.setVal(idx, jointVelocityModel[m_subModel.getJointMapping()[idx]]);
+    }
+
+    if (!m_kinDyn.setRobotState(worldTBase, qj, baseVelocity, dqj, worldGravity))
+    {
+        blf::log()->error("{} Unable to set the robot state.", logPrefix);
+        return false;
+    }
+
+    return updateDynamicsVariableState();
+}
+
+bool RDE::SubModelKinDynWrapper::updateDynamicsVariableState()
+{
+    constexpr auto logPrefix = "[BipedalLocomotion::Estimators::SubModelKinDynWrapper::"
+                               "updateDynamicsVariableState]";
+
+    iDynTree::FreeFloatingMassMatrix idynMassMatrix(m_kinDyn.model());
+    if (!m_kinDyn.getFreeFloatingMassMatrix(idynMassMatrix))
+    {
+        blf::log()->error("{} Unable to get the mass matrix of the sub-model.", logPrefix);
+        return false;
+    }
+    m_massMatrix = iDynTree::toEigen(idynMassMatrix);
+
+    iDynTree::FreeFloatingGeneralizedTorques idynGenTor(m_kinDyn.model());
```

Comment:
Similarly, also ``idynMassMatrix`` and ``idynGenTor`` require memory allocation

Reply:
I should have removed all the dynamic memory allocations now.